tag-state/calculate-tag-state.py

```diff
@@ -121,4 +158,122 @@ def semver_sorted_dict(d: dict) -> str:
+            # all our images that need manifests have arm64/amd64 versions
+            create_manifest = subprocess.run(
+                [
+                    "docker",
+                    "manifest",
+                    "create",
+                    gcr_image,
+                    f"{gcr_image}-arm64",
+                    f"{gcr_image}-amd64",
+                ]
+            )
+            if create_manifest.returncode is 0:
+                annotate_manifest_arm = subprocess.run(
+                    [
+                        "docker",
+                        "manifest",
+                        "annotate",
+                        gcr_image,
+                        f"{gcr_image}-arm64",
+                        "--arch",
+                        "arm64",
+                    ]
+                )
+                annotate_manifest_amd = subprocess.run(
+                    [
+                        "docker",
+                        "manifest",
+                        "annotate",
+                        gcr_image,
+                        f"{gcr_image}-amd64",
+                        "--arch",
+                        "amd64",
+                    ]
+                )
+                # if creation and annotation worked, push the new manifest
+                if (
+                    annotate_manifest_arm.returncode is 0
+                    and annotate_manifest_amd.returncode is 0
+                ):
+                    # rebuilds aren't a factor now and the tags are semver sorted, so the last writer is the right one
+                    major_manifest_name = f"gcr.io/ironcore-images/{image_name}:{major}"
+                    minor_manifest_name = (
+                        f"gcr.io/ironcore-images/{image_name}:{major}.{minor}"
+                    )
+                    subprocess.run(
+                        [
+                            "docker",
+                            "manifest",
+                            "create",
+                            major_manifest_name,
+                            f"{gcr_image}-arm64",
+                            f"{gcr_image}-amd64",
+                        ]
+                    )
+                    subprocess.run(
+                        [
+                            "docker",
+                            "manifest",
+                            "create",
+                            minor_manifest_name,
+                            f"{gcr_image}-arm64",
+                            f"{gcr_image}-amd64",
+                        ]
+                    )
+                    # the more specific annotation of the first manifest seems to carry through to the others
+                    subprocess.run(["docker", "manifest", "push", gcr_image])
+                    subprocess.run(["docker", "manifest", "push", major_manifest_name])
+                    subprocess.run(["docker", "manifest", "push", minor_manifest_name])
```

Comment:
Do you need to check the `returncode` on these subprocesses?

Reply:
"need" is a bit of a loaded term. No, since it's the last thing it'll try to do and the output will be in the some place I'd put it if it were to fail.